// ngx-extended-pdf-viewer/esm5/lib/toolbar/pdf-paging-area/pdf-last-page/pdf-last-page.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { PDFNotificationService } from './../../../pdf-notification-service';
import { Component, ViewChild, ElementRef } from '@angular/core';
var PdfLastPageComponent = /** @class */ (function () {
    function PdfLastPageComponent(notificationService) {
        var _this = this;
        this.notificationService = notificationService;
        this.disableLastPage = true;
        /** @type {?} */
        var subscription = this.notificationService.onPDFJSInit.subscribe((/**
         * @return {?}
         */
        function () {
            _this.onPdfJsInit();
            subscription.unsubscribe();
        }));
    }
    /**
     * @return {?}
     */
    PdfLastPageComponent.prototype.firstPage = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var PDFViewerApplication = ((/** @type {?} */ (window))).PDFViewerApplication;
        PDFViewerApplication.eventBus.dispatch('firstpage');
    };
    /**
     * @return {?}
     */
    PdfLastPageComponent.prototype.onPdfJsInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        /** @type {?} */
        var PDFViewerApplication = ((/** @type {?} */ (window))).PDFViewerApplication;
        PDFViewerApplication.eventBus.on('updateuistate', (/**
         * @param {?} event
         * @return {?}
         */
        function (event) { return _this.updateUIState(event); }));
    };
    /**
     * @param {?} event
     * @return {?}
     */
    PdfLastPageComponent.prototype.updateUIState = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        this.disableLastPage = event.pageNumber === event.pagesCount;
        this.button.nativeElement.disabled = this.disableLastPage;
    };
    /**
     * @return {?}
     */
    PdfLastPageComponent.prototype.lastPage = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var PDFViewerApplication = ((/** @type {?} */ (window))).PDFViewerApplication;
        PDFViewerApplication.eventBus.dispatch('lastpage');
    };
    PdfLastPageComponent.decorators = [
        { type: Component, args: [{
                    selector: 'pdf-last-page',
                    template: "<button type=\"button\"\n  class=\"toolbarButton hiddenLargeView\"\n  title=\"Last page\"\n  id=\"primaryLastPage\"\n  data-l10n-id=\"last_page\"\n  (click)=\"lastPage()\"\n  [disabled]=\"disableLastPage\"\n  #button\n  >\n  <svg style=\"width:24px;height:24px\" viewBox=\"0 0 24 24\">\n    <path fill=\"currentColor\" d=\"M5.59,7.41L10.18,12L5.59,16.59L7,18L13,12L7,6L5.59,7.41M16,6H18V18H16V6Z\" />\n  </svg>\n  <span data-l10n-id=\"last_page_label\">Last page</span>\n</button>\n",
                    styles: [""]
                }] }
    ];
    /** @nocollapse */
    PdfLastPageComponent.ctorParameters = function () { return [
        { type: PDFNotificationService }
    ]; };
    PdfLastPageComponent.propDecorators = {
        button: [{ type: ViewChild, args: ['button',] }]
    };
    return PdfLastPageComponent;
}());
export { PdfLastPageComponent };
if (false) {
    /** @type {?} */
    PdfLastPageComponent.prototype.disableLastPage;
    /**
     * @type {?}
     * @private
     */
    PdfLastPageComponent.prototype.button;
    /**
     * @type {?}
     * @private
     */
    PdfLastPageComponent.prototype.notificationService;
}
