// ngx-extended-pdf-viewer/esm2015/lib/toolbar/pdf-zoom-toolbar/pdf-zoom-toolbar.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, Output, EventEmitter } from '@angular/core';
export class PdfZoomToolbarComponent {
    constructor() {
        this.showZoomButtons = true;
        this.zoomChange = new EventEmitter();
    }
    /**
     * @param {?} event
     * @return {?}
     */
    emitZoomChange(event) {
        this.zoomChange.emit(event);
    }
}
PdfZoomToolbarComponent.decorators = [
    { type: Component, args: [{
                selector: 'pdf-zoom-toolbar',
                template: "<div [class.invisible]=\"!showZoomButtons\" id=\"toolbarViewerMiddle\">\n  <div class=\"splitToolbarButton\">\n    <pdf-zoom-out></pdf-zoom-out>\n    <div class=\"splitToolbarButtonSeparator\"></div>\n    <pdf-zoom-in></pdf-zoom-in>\n  </div>\n  <pdf-zoom-dropdown\n      style=\"width: unset; max-width: unset;padding-top:3px\"\n      (zoomChange)=\"emitZoomChange($event)\"></pdf-zoom-dropdown>\n</div>\n",
                styles: [""]
            }] }
];
PdfZoomToolbarComponent.propDecorators = {
    showZoomButtons: [{ type: Input }],
    zoomChange: [{ type: Output }]
};
if (false) {
    /** @type {?} */
    PdfZoomToolbarComponent.prototype.showZoomButtons;
    /** @type {?} */
    PdfZoomToolbarComponent.prototype.zoomChange;
}
