// ngx-extended-pdf-viewer/esm5/lib/options/pdf-viewer-application.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * @record
 */
export function IPDFViewerApplication() { }
if (false) {
    /** @type {?} */
    IPDFViewerApplication.prototype.appConfig;
    /** @type {?} */
    IPDFViewerApplication.prototype._boundEvents;
    /** @type {?} */
    IPDFViewerApplication.prototype.enablePrint;
    /** @type {?} */
    IPDFViewerApplication.prototype.eventBus;
    /** @type {?} */
    IPDFViewerApplication.prototype.isViewerEmbedded;
    /** @type {?} */
    IPDFViewerApplication.prototype.onError;
    /** @type {?} */
    IPDFViewerApplication.prototype.page;
    /** @type {?} */
    IPDFViewerApplication.prototype.pagesCount;
    /** @type {?} */
    IPDFViewerApplication.prototype.pdfDocument;
    /** @type {?} */
    IPDFViewerApplication.prototype.pdfLinkService;
    /** @type {?} */
    IPDFViewerApplication.prototype.pdfSidebar;
    /** @type {?} */
    IPDFViewerApplication.prototype.pdfViewer;
    /** @type {?} */
    IPDFViewerApplication.prototype.printKeyDownListener;
    /** @type {?} */
    IPDFViewerApplication.prototype.sidebarViewOnLoad;
    /** @type {?} */
    IPDFViewerApplication.prototype.spreadModeOnLoad;
    /**
     * @return {?}
     */
    IPDFViewerApplication.prototype.cleanup = function () { };
    /**
     * @return {?}
     */
    IPDFViewerApplication.prototype.close = function () { };
    /**
     * @param {?} source
     * @param {?=} options
     * @return {?}
     */
    IPDFViewerApplication.prototype.open = function (source, options) { };
    /**
     * @return {?}
     */
    IPDFViewerApplication.prototype.unbindEvents = function () { };
    /**
     * @return {?}
     */
    IPDFViewerApplication.prototype.unbindWindowEvents = function () { };
}
