// ngx-extended-pdf-viewer/esm5/lib/toolbar/pdf-find-button/pdf-find-button.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
var PdfFindButtonComponent = /** @class */ (function () {
    function PdfFindButtonComponent() {
        this.showFindButton = undefined;
        this.textLayer = undefined;
    }
    /**
     * @return {?}
     */
    PdfFindButtonComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    PdfFindButtonComponent.decorators = [
        { type: Component, args: [{
                    selector: 'pdf-find-button',
                    template: "<button\n  type=\"button\"\n  [class.invisible]=\"!showFindButton || !textLayer\"\n  id=\"viewFind\"\n  class=\"toolbarButton\"\n  title=\"Find in Document\"\n  data-l10n-id=\"findbar\"\n>\n  <svg style=\"width:24px;height:24px\" viewBox=\"0 0 24 24\">\n    <path fill=\"currentColor\" d=\"M9.5,3A6.5,6.5 0 0,1 16,9.5C16,11.11 15.41,12.59 14.44,13.73L14.71,14H15.5L20.5,19L19,20.5L14,15.5V14.71L13.73,14.44C12.59,15.41 11.11,16 9.5,16A6.5,6.5 0 0,1 3,9.5A6.5,6.5 0 0,1 9.5,3M9.5,5C7,5 5,7 5,9.5C5,12 7,14 9.5,14C12,14 14,12 14,9.5C14,7 12,5 9.5,5Z\" />\n  </svg>\n  <span data-l10n-id=\"findbar_label\">Find</span>\n</button>\n",
                    styles: ["button svg{margin-top:-2px}:host:focus{outline:0}button:focus,svg:focus{outline:0}"]
                }] }
    ];
    /** @nocollapse */
    PdfFindButtonComponent.ctorParameters = function () { return []; };
    PdfFindButtonComponent.propDecorators = {
        showFindButton: [{ type: Input }],
        textLayer: [{ type: Input }]
    };
    return PdfFindButtonComponent;
}());
export { PdfFindButtonComponent };
if (false) {
    /** @type {?} */
    PdfFindButtonComponent.prototype.showFindButton;
    /** @type {?} */
    PdfFindButtonComponent.prototype.textLayer;
}
