// ngx-extended-pdf-viewer/esm5/lib/toolbar/pdf-findbar/pdf-findbar.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, TemplateRef } from '@angular/core';
var PdfFindbarComponent = /** @class */ (function () {
    function PdfFindbarComponent() {
        this.showFindButton = true;
    }
    PdfFindbarComponent.decorators = [
        { type: Component, args: [{
                    selector: 'pdf-findbar',
                    template: "<ng-container [ngTemplateOutlet]=\"customFindbar ? customFindbar : defaultFindbar\">\n</ng-container>\n\n<ng-template #defaultFindbar>\n  <div\n    class=\"findbar hidden doorHanger\"\n    [class.invisible]=\"!showFindButton\"\n    id=\"findbar\"\n    [style.transform]=\"'scale(' + mobileFriendlyZoomScale + ')'\"\n    [style.transformOrigin]=\"'left center'\"\n    [style.left]=\"findbarLeft\"\n    [style.top]=\"findbarTop\"\n  >\n    <ng-container [ngTemplateOutlet]=\"customFindbarButtons ? customFindbarButtons : defaultFindbarButtons\"> </ng-container>\n    </div>\n</ng-template>\n\n<ng-template #defaultFindbarButtons>\n  <pdf-find-input-area [customFindbarInputArea]=\"customFindbarInputArea\"></pdf-find-input-area>\n  <pdf-findbar-options-one-container></pdf-findbar-options-one-container>\n  <pdf-findbar-options-two-container></pdf-findbar-options-two-container>\n  <pdf-findbar-options-three-container></pdf-findbar-options-three-container>\n  <pdf-findbar-message-container></pdf-findbar-message-container>\n</ng-template>\n",
                    styles: [""]
                }] }
    ];
    PdfFindbarComponent.propDecorators = {
        showFindButton: [{ type: Input }],
        mobileFriendlyZoomScale: [{ type: Input }],
        findbarLeft: [{ type: Input }],
        findbarTop: [{ type: Input }],
        customFindbarInputArea: [{ type: Input }],
        customFindbar: [{ type: Input }],
        customFindbarButtons: [{ type: Input }]
    };
    return PdfFindbarComponent;
}());
export { PdfFindbarComponent };
if (false) {
    /** @type {?} */
    PdfFindbarComponent.prototype.showFindButton;
    /** @type {?} */
    PdfFindbarComponent.prototype.mobileFriendlyZoomScale;
    /** @type {?} */
    PdfFindbarComponent.prototype.findbarLeft;
    /** @type {?} */
    PdfFindbarComponent.prototype.findbarTop;
    /** @type {?} */
    PdfFindbarComponent.prototype.customFindbarInputArea;
    /** @type {?} */
    PdfFindbarComponent.prototype.customFindbar;
    /** @type {?} */
    PdfFindbarComponent.prototype.customFindbarButtons;
}
