// ngx-extended-pdf-viewer/esm5/lib/toolbar/pdf-toolbar/pdf-toolbar.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, TemplateRef, Output, EventEmitter } from '@angular/core';
var PdfToolbarComponent = /** @class */ (function () {
    function PdfToolbarComponent() {
        this.mobileFriendlyZoomScale = 1;
        this.primaryMenuVisible = true;
        this.showBookmarkButton = true;
        this.showDownloadButton = true;
        this.showFindButton = undefined;
        this.showHandToolButton = true;
        this.showOpenFileButton = true;
        this.showPrintButton = true;
        this.showPagingButtons = true;
        this.showPresentationModeButton = false;
        this.showRotateButton = true;
        this.showSecondaryToolbarButton = true;
        this.showSidebarButton = true;
        this.showZoomButtons = true;
        this.textLayer = undefined;
        this.toolbarPaddingTop = '0px';
        this.toolbarWidth = '100%';
        this.zoomChange = new EventEmitter();
    }
    /**
     * @return {?}
     */
    PdfToolbarComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () { };
    /**
     * @param {?} value
     * @return {?}
     */
    PdfToolbarComponent.prototype.emitZoomChange = /**
     * @param {?} value
     * @return {?}
     */
    function (value) {
        this.zoomChange.emit(value);
    };
    PdfToolbarComponent.decorators = [
        { type: Component, args: [{
                    selector: 'pdf-toolbar',
                    template: "<div class=\"toolbar\" [class.invisible]=\"!primaryMenuVisible\" >\n  <div\n    id=\"toolbarContainer\"\n    [style.transform]=\"'scale(' + mobileFriendlyZoomScale + ')'\"\n    [style.transformOrigin]=\"'left center'\"\n    [style.width]=\"toolbarWidth\"\n    [style.paddingTop]=\"toolbarPaddingTop\"\n  >\n    <ng-content *ngTemplateOutlet=\"customToolbar ? customToolbar : defaultToolbar\"></ng-content>\n    <div id=\"loadingBar\">\n      <div class=\"progress\"><div class=\"glimmer\"></div></div>\n    </div>\n  </div>\n</div>\n\n<ng-template #defaultToolbar>\n  <div id=\"toolbarViewer\">\n    <div id=\"toolbarViewerLeft\">\n      <pdf-toggle-sidebar [showSidebarButton]=\"showSidebarButton\"></pdf-toggle-sidebar>\n      <pdf-find-button [showFindButton]=\"showFindButton\" [textLayer]=\"textLayer\"></pdf-find-button>\n      <pdf-paging-area [showPagingButtons]=\"showPagingButtons\"></pdf-paging-area>\n    </div>\n    <pdf-zoom-toolbar [showZoomButtons]=\"showZoomButtons\" (zoomChange)=\"emitZoomChange($event)\"></pdf-zoom-toolbar>\n    <div id=\"toolbarViewerRight\">\n      <pdf-hand-tool [showHandToolButton]=\"showHandToolButton\"></pdf-hand-tool>\n      <pdf-select-tool [showSelectToolButton]=\"showHandToolButton\"></pdf-select-tool>\n      <pdf-rotate-page [showRotateButton]=\"showRotateButton\"></pdf-rotate-page>\n      <pdf-presentation-mode [showPresentationModeButton]=\"showPresentationModeButton\"></pdf-presentation-mode>\n      <pdf-open-file [showOpenFileButton]=\"showOpenFileButton\"></pdf-open-file>\n      <pdf-print [showPrintButton]=\"showPrintButton\"></pdf-print>\n      <pdf-download [showDownloadButton]=\"showDownloadButton\"></pdf-download>\n      <pdf-bookmark [showBookmarkButton]=\"showBookmarkButton\"></pdf-bookmark>\n      <div [class.invisible]=\"!showSecondaryToolbarButton\" class=\"verticalToolbarSeparator hiddenSmallView\"></div>\n      <pdf-toggle-secondary-toolbar [showSecondaryToolbarButton]=\"showSecondaryToolbarButton\"></pdf-toggle-secondary-toolbar>\n    </div>\n   </div>\n</ng-template>\n",
                    styles: [""]
                }] }
    ];
    /** @nocollapse */
    PdfToolbarComponent.ctorParameters = function () { return []; };
    PdfToolbarComponent.propDecorators = {
        customToolbar: [{ type: Input }],
        mobileFriendlyZoomScale: [{ type: Input }],
        primaryMenuVisible: [{ type: Input }],
        showBookmarkButton: [{ type: Input }],
        showDownloadButton: [{ type: Input }],
        showFindButton: [{ type: Input }],
        showHandToolButton: [{ type: Input }],
        showOpenFileButton: [{ type: Input }],
        showPrintButton: [{ type: Input }],
        showPagingButtons: [{ type: Input }],
        showPresentationModeButton: [{ type: Input }],
        showRotateButton: [{ type: Input }],
        showSecondaryToolbarButton: [{ type: Input }],
        showSidebarButton: [{ type: Input }],
        showZoomButtons: [{ type: Input }],
        textLayer: [{ type: Input }],
        toolbarPaddingTop: [{ type: Input }],
        toolbarWidth: [{ type: Input }],
        zoomChange: [{ type: Output }]
    };
    return PdfToolbarComponent;
}());
export { PdfToolbarComponent };
if (false) {
    /** @type {?} */
    PdfToolbarComponent.prototype.customToolbar;
    /** @type {?} */
    PdfToolbarComponent.prototype.mobileFriendlyZoomScale;
    /** @type {?} */
    PdfToolbarComponent.prototype.primaryMenuVisible;
    /** @type {?} */
    PdfToolbarComponent.prototype.showBookmarkButton;
    /** @type {?} */
    PdfToolbarComponent.prototype.showDownloadButton;
    /** @type {?} */
    PdfToolbarComponent.prototype.showFindButton;
    /** @type {?} */
    PdfToolbarComponent.prototype.showHandToolButton;
    /** @type {?} */
    PdfToolbarComponent.prototype.showOpenFileButton;
    /** @type {?} */
    PdfToolbarComponent.prototype.showPrintButton;
    /** @type {?} */
    PdfToolbarComponent.prototype.showPagingButtons;
    /** @type {?} */
    PdfToolbarComponent.prototype.showPresentationModeButton;
    /** @type {?} */
    PdfToolbarComponent.prototype.showRotateButton;
    /** @type {?} */
    PdfToolbarComponent.prototype.showSecondaryToolbarButton;
    /** @type {?} */
    PdfToolbarComponent.prototype.showSidebarButton;
    /** @type {?} */
    PdfToolbarComponent.prototype.showZoomButtons;
    /** @type {?} */
    PdfToolbarComponent.prototype.textLayer;
    /** @type {?} */
    PdfToolbarComponent.prototype.toolbarPaddingTop;
    /** @type {?} */
    PdfToolbarComponent.prototype.toolbarWidth;
    /** @type {?} */
    PdfToolbarComponent.prototype.zoomChange;
}
