// ngx-extended-pdf-viewer/esm5/lib/sidebar/pdf-sidebar/pdf-sidebar.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
var PdfSidebarComponent = /** @class */ (function () {
    function PdfSidebarComponent() {
        this.showSidebarButton = true;
    }
    PdfSidebarComponent.decorators = [
        { type: Component, args: [{
                    selector: 'pdf-sidebar',
                    template: "<div id=\"sidebarContainer\" [style.top]=\"sidebarPositionTop\">\n  <div\n    id=\"additionalSidebarContainer\"\n    [class.invisible]=\"!showSidebarButton\"\n  >\n    <div id=\"toolbarSidebar\">\n      <div class=\"splitToolbarButton toggled\">\n        <button type=\"button\"\n          id=\"viewThumbnail\"\n          class=\"toolbarButton toggled\"\n          title=\"Show Thumbnails\"\n          data-l10n-id=\"thumbs\"\n        >\n          <span data-l10n-id=\"thumbs_label\">Thumbnails</span>\n        </button>\n        <button type=\"button\"\n          id=\"viewOutline\"\n          class=\"toolbarButton\"\n          title=\"Show Document Outline (double-click to expand/collapse all items)\"\n          data-l10n-id=\"document_outline\"\n        >\n          <span data-l10n-id=\"document_outline_label\"\n            >Document Outline</span\n          >\n        </button>\n        <button type=\"button\"\n          id=\"viewAttachments\"\n          class=\"toolbarButton\"\n          title=\"Show Attachments\"\n          data-l10n-id=\"attachments\"\n        >\n          <span data-l10n-id=\"attachments_label\">Attachments</span>\n        </button>\n      </div>\n    </div>\n  </div>\n  <div id=\"sidebarContent\">\n    <div id=\"thumbnailView\"></div>\n    <div id=\"outlineView\" class=\"hidden\"></div>\n    <div id=\"attachmentsView\" class=\"hidden\"></div>\n  </div>\n  <div id=\"sidebarResizer\" class=\"hidden\"></div>\n</div>\n",
                    styles: [""]
                }] }
    ];
    PdfSidebarComponent.propDecorators = {
        sidebarPositionTop: [{ type: Input }],
        showSidebarButton: [{ type: Input }]
    };
    return PdfSidebarComponent;
}());
export { PdfSidebarComponent };
if (false) {
    /** @type {?} */
    PdfSidebarComponent.prototype.sidebarPositionTop;
    /** @type {?} */
    PdfSidebarComponent.prototype.showSidebarButton;
}
