// ngx-extended-pdf-viewer/esm5/lib/toolbar/pdf-paging-area/page-number/page-number.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
var PageNumberComponent = /** @class */ (function () {
    function PageNumberComponent() {
        this.showPagingButtons = true;
    }
    /**
     * @return {?}
     */
    PageNumberComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () { };
    PageNumberComponent.decorators = [
        { type: Component, args: [{
                    selector: 'page-number',
                    template: "<input\n  [class.invisible]=\"!showPagingButtons\"\n  type=\"number\"\n  id=\"pageNumber\"\n  class=\"toolbarField pageNumber\"\n  title=\"Page\"\n  value=\"1\"\n  size=\"4\"\n  min=\"1\"\n  data-l10n-id=\"page\"\n/>\n<span [class.invisible]=\"!showPagingButtons\" id=\"numPages\" class=\"toolbarLabel\"></span>\n",
                    styles: [""]
                }] }
    ];
    /** @nocollapse */
    PageNumberComponent.ctorParameters = function () { return []; };
    PageNumberComponent.propDecorators = {
        showPagingButtons: [{ type: Input }]
    };
    return PageNumberComponent;
}());
export { PageNumberComponent };
if (false) {
    /** @type {?} */
    PageNumberComponent.prototype.showPagingButtons;
}
